app/scripts/controllers/overview.js

```diff
@@ -188,12 +193,13 @@ angular.module('openshiftConsole')
     };
 
     var groupPods = function() {
-      if (!pods || !deployments) {
+      if (!pods || !deployments || !replicaSets) {
         return;
       }
 
-      $scope.podsByDeployment = PodsService.groupByReplicationController(pods, deployments);
-      $scope.monopodsByService = PodsService.groupByService($scope.podsByDeployment[''], services, showMonopod);
+      var allOwners = _.toArray(deployments).concat(_.toArray(replicaSets));
```

Comment:
I did that originally. This seemed cleaner though and it has the same time complexity since we iterate over the list in the service anyway.